data_diff/databases/exasol.py

```diff
@@ -0,0 +1,161 @@
+import re
+
+from .database_types import *
+from .base import ThreadedDatabase, import_helper, ConnectError, QueryError
+from .base import (
+    MD5_HEXDIGITS,
+    CHECKSUM_HEXDIGITS,
+    TIMESTAMP_PRECISION_POS,
+    DEFAULT_DATETIME_PRECISION,
+)
+
+@import_helper("exasol")
+def import_exasol():
+    import pyexasol
+
+    return pyexasol
+
+SESSION_TIME_ZONE = None  # Changed by the tests
+
+def _query_conn(conn, sql_code: str) -> list:
+    c = conn.execute(sql_code)
+    if sql_code.lower().startswith("select"):
+        return c.fetchall()
+
+
+class Exasol(ThreadedDatabase):
+    TYPE_CLASSES: Dict[str, type] = {
+        "DECIMAL": Decimal,
+        "DOUBLE PRECISSION": Float,
+        # Text
+        "CHAR": Text,
+    }
+    ROUNDS_ON_PREC_LOSS = True
+
+    def __init__(self, *, host, thread_count, **kw):
+        self.kwargs = dict(dsn="%s" % host, **kw)
+
+        self.default_schema = kw.get("schema")
+
+        super().__init__(thread_count=thread_count)
+
+    def create_connection(self):
+        self._exasol = import_exasol()
+        try:
+            c = self._exasol.connect(**self.kwargs)
+            if SESSION_TIME_ZONE:
+                c.execute(f"ALTER SESSION SET TIME_ZONE = '{SESSION_TIME_ZONE}'")
+            return c
+        except Exception as e:
```

Comment:
Please catch an exasol base exception here, not just every exception.

Reply:
Fixed in a5c90dd